sdk/storage/azure-storage-blob/azure/storage/blob/aio/download_async.py

```diff
@@ -369,37 +374,44 @@ def __aiter__(self):
             The number of parallel connections with which to download.
         :rtype: bytes
         """
-        stream = BytesIO()
-        await self.download_to_stream(stream, max_concurrency=max_concurrency)
-        return stream.getvalue()
+        warnings.warn(
+            "content_as_bytes is deprecated, use read_all instead",
+            DeprecationWarning
+        )
+        self._max_concurrency = max_concurrency
+        return await self.readall()
 
-    async def content_as_text(self, max_concurrency=1, encoding='UTF-8'):
-        """Download the contents of this file, and decode as text.
+    async def content_as_text(self, max_concurrency=1, encoding="UTF-8"):
```

Comment:
Weren't we going to remove `content_as_text` and provide the optional encoding param to `content_as_bytes` itself?

Reply:
I haven't removed or changed any of the existing functions - they've just been deprecated.
When using the _new_ readall function, you have to pass in encoding as a kwarg if you want the decoding behaviour.